start.sh

```diff
@@ -0,0 +1,243 @@
+#!/bin/bash
+set -e -o pipefail
+
+# default values which can be overriden by -f or -p flags
+COMPOSE_CFG=
+PREFIX=tb
+
+usage() {
+    echo 'Usage ./start.sh [-f docker-compose.yml] [-p project] [--no-index] [cmd] [args]'
+    echo
+    echo 'Starts up the entire stack.'
+    echo
+    echo '-f <filename> [optional] use this file as the docker-compose config file'
+    echo '-p <project>  [optional] use this name as the project prefix for docker-compose'
+    echo '-h            help. print this thing you are reading now.'
+    echo
+    echo 'Optionally pass a command and parameters and this script will execute just'
+    echo 'that command, for testing purposes.'
+}
+
+env() {
+    if [ ! -f ".env" ]; then
+        echo 'You need a configuration file with credentials.'
+        echo 'Copying .env.example to .env'
+        cp .env.example .env
+        cat .env.example
+        exit 1
+    else
+        . .env
+    fi
+    CB_USER=${CB_USER:-Administrator}
+    CB_PASSWORD=${CB_PASSWORD:-password}
+    CB_RAM_QUOTA=${CB_RAM_QUOTA:-100}
+}
+
+prep() {
+    echo "Starting example application"
+    echo "project prefix:      $PREFIX"
+    echo "docker-compose file: $CONFIG_FILE"
+    echo
+    echo 'Pulling latest container images'
+    ${COMPOSE} pull
+}
+
+# get the IP:port of a container via either the local docker-machine or from
+# sdc-listmachines.
+getIpPort() {
+    if [ -z "${COMPOSE_CFG}" ]; then
+        local ip=$(sdc-listmachines --name ${PREFIX}_$1_1 | json -a ips.1)
+        local port=$2
+    else
+        local ip=$(docker-machine ip default)
+        local port=$(docker inspect ${PREFIX}_$1_1 | json -a NetworkSettings.Ports."$2/tcp".0.HostPort)
+    fi
+    echo "$ip:$port"
+}
+
+# start and initialize the Couchbase cluster, along with Consul
+startDatabase() {
+    echo
+    echo 'Starting Couchbase'
+    ${COMPOSE} up -d --no-recreate couchbase
```

Comment:
If we don't start Couchbase first we're at risk of a race condition between the first node of Couchbase being started and registered with Consul, and the first Touchbase node coming up. The first CB node needs to be up and registered because we're relying on the CB client to be able to access it (because TB doesn't support graceful reload).